flink-table/flink-table-planner/src/main/java/org/apache/flink/table/executor/StreamingExecutor.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.executor;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.api.common.JobExecutionResult;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.transformations.StreamTransformation;
+
+import java.util.List;
+
+/**
+ * An implementation of {@link Executor} that is backed by a {@link StreamExecutionEnvironment}.
+ * This is the only executor that {@link org.apache.flink.table.planner.StreamPlanner} supports.
+ */
+@Internal
+public class StreamingExecutor implements Executor {
+	private final StreamExecutionEnvironment executionEnvironment;
+
+	StreamingExecutor(StreamExecutionEnvironment executionEnvironment) {
+		this.executionEnvironment = executionEnvironment;
+	}
+
+	@Override
+	public void apply(List<StreamTransformation<?>> transformations) {
+		transformations.forEach(executionEnvironment::addOperator);
+	}
+
+	@Override
+	public JobExecutionResult execute(String jobName) throws Exception {
+		return executionEnvironment.execute(jobName);
```

Comment:
there is a corner case: if an exception is thrown in `getStreamGraph` method , the `transformations` does not be cleared in `execute` method. and if the job is re-submitted, the DAG is unexcepted due to residual `transformations`.